homeassistant/components/lektrico/coordinator.py

```diff
@@ -0,0 +1,51 @@
+"""Coordinator for the Lektrico Charging Station integration."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+
+from aiohttp import ClientSession
+from lektricowifi import DeviceConnectionError, lektricowifi
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import LOGGER
+
+SCAN_INTERVAL = timedelta(seconds=10)
+
+
+class LektricoDeviceDataUpdateCoordinator(DataUpdateCoordinator[lektricowifi.Info]):
+    """Data update coordinator for Lektrico device."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        friendly_name: str,
+        host: str,
+        session: ClientSession,
+        serial_number: str,
+        board_revision: str,
+        device_type: str,
```

Comment:
You can access `self.config_entry` after `super().__init__()` has been called in the coordinator. But you have to overwrite the type. So if you put `config_entry: ConfigEntry` at class level, you can just access `self.config_entry` here instead

Reply:
I added  `config_entry: ConfigEntry` and removed parameters from __init__, thanks.